tornado/test/websocket_test.py

```diff
@@ -329,6 +336,13 @@ def test_check_origin_invalid_subdomains(self):
 
         self.assertEqual(cm.exception.code, 403)
 
+    @gen_test
+    def test_handshake_response_headers(self):
+        ws = yield self.ws_connect('/handshake_response_headers')
+
+        self.assertEqual(ws.headers.get('Set-Cookie'), 'hello=1')
```

Comment:
PR updated to not introduce new interface.
What do you think on using `set_default_headers` method for this purpose?